src/api/api.py

```diff
@@ -19,6 +19,18 @@ def get_access_data(username: str) -> Optional[dict]:
         return {}
 
 
+def get_stats_data(username: str) -> Optional[dict]:
+    try:
+        stats_table = Table(airtable_key, base_id, "Stats")
+        formula = match({"Username": username})
+        # Dict record if exists, None otherwises
+        record = stats_table.first(formula=formula)
+        return record
+
+    except RequestException:
+        return {}
+
+
```

Comment:
Can we make a reusable function for fetching table data? Would something like this work?

```suggestion
def get_stats_data(username: str, table_name: str) -> Optional[dict]:
    try:
        stats_table = Table(airtable_key, base_id, table_name)
        formula = match({"Username": username})
        # Dict record if exists, None otherwises
        record = stats_table.first(formula=formula)
        return record

    except RequestException:
        return {}


```

Reply:
Done